packages/frontend/src/page/page_container.tsx

```diff
@@ -17,9 +18,12 @@ export function PageContainer(props: {
     const [loginOpen, setLoginOpen] = createSignal(false);
     const [openImport, setImportOpen] = createSignal(false);
 
+    const deleteDocument = useDeleteDocument();
```

Comment:
Encapsulating everything needed for the delete dialog in the function `useDeleteDocument` is cleaner than how we're handling the login and import dialogs in this component. In the future, we should follow your pattern in those cases too.